src/emulator/instruction.ts

```diff
@@ -470,17 +533,18 @@ apply(0x28, { op: Operation.jr, mode1: AddressingMode.imm8, condition: Condition
 apply(0x30, { op: Operation.jr, mode1: AddressingMode.imm8, condition: Condition.nc, cycles: 2, len: 2 });
 apply(0x38, { op: Operation.jr, mode1: AddressingMode.imm8, condition: Condition.c, cycles: 2, len: 2 });
 
-apply(0xf3, { op: Operation.di, cycles: 1, len: 1 });
-apply(0xfb, { op: Operation.ei, cycles: 1, len: 1 });
-
-apply(0xcd, { op: Operation.call, mode1: AddressingMode.imm16, cycles: 8, len: 3 });
+apply(0xcd, { op: Operation.call, mode1: AddressingMode.imm16, cycles: 6, len: 3 });
+apply(0xc4, { op: Operation.call, mode1: AddressingMode.imm16, condition: Condition.nz, cycles: 3, len: 3 });
+apply(0xd4, { op: Operation.call, mode1: AddressingMode.imm16, condition: Condition.z, cycles: 3, len: 3 });
+apply(0xcc, { op: Operation.call, mode1: AddressingMode.imm16, condition: Condition.nc, cycles: 3, len: 3 });
+apply(0xdc, { op: Operation.call, mode1: AddressingMode.imm16, condition: Condition.c, cycles: 3, len: 3 });
 
 apply(0xc9, { op: Operation.ret, cycles: 4, len: 1 });
```

Comment:
See above; I think we should adjust cycles to 2 here and apply the same logic to all `RET` variants in `cpu.ts`.

Reply:
had cycles adjusted to 21, because RET has only 4 cycles and the conditional RET 5 if they jump